packages/devtools_shared/lib/src/common.dart

```diff
@@ -2,5 +2,20 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'package:path/path.dart' as path;
+
 /// Describes an instance of the Dart Tooling Daemon.
 typedef DTDConnectionInfo = ({String? uri, String? secret});
+
+/// The name of the Directory where a Dart application's package config file is
+/// stored.
+const dartToolDirectoryName = '.dart_tool';
+
+/// The name of the package config file for a Dart application.
+const packageConfigFileName = 'package_config.json';
+
+/// The path identifier for the package config URI for a Dart application.
+///
+/// The package config file lives at '.dart_tool/package_config.json'.
+final packageConfigIdentifier =
```

Comment:
Note that the name/location of this file is not guaranteed to be consistent. This is just the default path.

Reply:
Consider also the pub workspaces feature, I am not sure if this will work well with that or not.

But, in the future it will be common externally to have a `.dart_tool/package_config.json` file which lives in a directory above the actual package root (not sure if that will matter for this or not).

Reply:
What is an example where the package config will not be named `package_config.json`? It is my understanding that much of our tooling system, pub, etc. rely on this assumption.

Even with a pub workspace, shouldn't the URI to the package config still end with `.dart_tool/package_config.json`? @sigurdm 

Generally, I filed https://github.com/flutter/devtools/issues/7944 to track ensuring DevTools extensions work well with pub workspaces.

Reply:
Any user or tool for any reason can pass any path they want for the `--packages` argument. Yes, most of the time this doesn't happen, but it is supported.

Specifically, build systems might create a file by any name, and pass the path to it that way.

Reply:
> Even with a pub workspace, shouldn't the URI to the package config still end with .dart_tool/package_config.json? @sigurdm

Yes with workspaces it will - but also what @jakemac53 said!

Reply:
I have removed these shared constants and filed https://github.com/flutter/devtools/issues/7944 to track moving away from the assumptions around `.dart_tool/package_config.json`

Reply:
It is also worth noting that in the workspace case, the package config will contain all packages depended on by any package in the workspace, not just the current package.

I don't know exactly all the details here, but if DevTools is looking in that file for certain dependencies, assuming they are dependencies of the current package, then that assumption won't hold up (it will contain dependencies which are not transitive deps of the current package).

I don't know enough to suggest a fix for that though.